### ISSUE TITLE
node downloads without npm for example i put node -v i get 8.9.3 but when i type npm -v i get nothing... also when i type npm install with the bot I'm using for sneakers it shows command not found.. any help?

### ISSUE THREAD

**Issue by techman556** [nvm-windows/nvm#329]
If this is a question about how to use NVM4W, please use [stackoverflow](http://stackoverflow.com/questions/tagged/nvm-windows) instead.

If this is an issue regarding antivirus, make sure you search the existing issues first.

### My Environment

- [ ] Windows 7 or below (not truly supported due to EOL - see wiki for details)
- [ ] Windows 8
- [ ] Windows 8.1
- [ ] Windows 10
- [ ] Windows 10 IoT Core
- [ ] Windows Server 2012
- [ ] Windows Server 2012 R2
- [ ] Windows Server 2016

- [ ] My Windows installation is non-English.

### I'm using NVM4W version:

- [ ] 1.1.6
- [ ] 1.1.5
- [ ] 1.1.4
- [ ] 1.1.3
- [ ] 1.1.2
- [ ] 1.1.1
- [ ] Older
- [ ] OTHER (Please Specify)

### I have already...

- [ ] read the README to be aware of npm gotchas & antivirus issues.
- [ ] reviewed the [wiki](https://github.com/coreybutler/nvm-windows/wiki) to make sure my issue hasn't already been resolved.
- [ ] verified I'm using an account with administrative privileges.
- [ ] searched the issues (open and closed) to make sure this isn't a duplicate.
- [ ] made sure this isn't a question about how to use NVM for Windows, since [gitter](https://gitter.im/coreybutler/nvm-windows?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge) is used for questions and comments.

### My issue is related to (check only those which apply):

- [ ] settings.txt
- [ ] proxy support (Have you tried version 1.1.0+?)
- [ ] 32 or 64 bit support (Have you tried version 1.1.3+?)
- [ ] Character escaping (Have you tried version 1.1.6+?)
- [ ] A standard shell environment (terminal/powershell)
- [ ] A non-standard shell environment (Cmder, Hyper, Cygwin, git)

### Expected Behavior

Fill me in...

### Actual Behavior

Fill me in...

### Steps to reproduce the problem:

Fill me in...

**Comment by maeriens** [comment]
Did you do `nvm use 8.9.3` after the installation? You need to select a node version before running npm and similar commands.

**Comment by techman556** [comment]
it just says command not found

**Comment by maeriens** [comment]
Could you reformat the issue a bit? The title should not be de description, make it a summary so it's easier for people having similar issues in the future - the description of the issue should be in the body of it.

Now, to the thing. To be clear, your steps were as follows:
 - installed `nvm-windows` from the [setup page](https://github.com/coreybutler/nvm-windows/releases)
 - opened the console, ran `nvm install 8.9.3`
 - It finished and now if you run `nvm use 8.9.3` it reads command not found? (Note I am talking about `nvm` not `npm`)
 - If so, have you tried removing `nvm-windows` and starting again?
 - If `nvm use 8.9.3` did output `Now using node v8.9.4`, can you try **then** running `npm -v`?

**Comment by techman556** [comment]
also comes command not found

**Comment by maeriens** [comment]
Could you confirm that you did the previous steps, download `nvm-windows`, installed that and not `node.js`? Have you [added npm to your PATH](https://stackoverflow.com/questions/20992723/npm-is-not-recognized-as-internal-or-external-command-operable-program-or-bat)?
What have you tried, what have you done, what is the expected output, what is the actual output?
Being as specific as possible helps us all here...

**Comment by techman556** [comment]
yes i know nvm. thanks for the help btw. I appreciate this. Im also running on mac
npm -v + command not found also nvm install 8.9.3 + command not found

**Comment by techman556** [comment]
i downloaded npm for nodes.org

**Comment by maeriens** [comment]
You are working on OSX or in Windows? This repository is `nvm-windows` - the package manager for **_Windows_**. If you are in OSX, you should check in other places for solutions.

**Comment by ferlolo** [comment]
Hi,
I have the same problem, when i execute nvm install v9.3.0 (for example) nvm create a temporal folder but not set npm module into node_modules of v9.3.0.
maybe are a permission problem but i don't know now, i will test, if someone can help us i would appreciate it.

My version of NVM is 1.1.6.

Regards

**Comment by systemundertestcode** [comment]
Hey guys, 
I had a similar problem on windows 10, and as another user points out above, once I added the nodejs exe that nvm installs to my PATH, it all works nicely!

**Comment by sparachi** [comment]
If I install 6.9.2 using nvm it works fine. If I install anything above that ( I tried both 8.9.2 and 9.4.0) it says the installation is succesfull. I also did `nvm use 8.9.2` . But nothing happens, If I use npm install in any repo, i get npm is not recognized as internal or external command.

**Comment by petevb** [comment]
I'm having the same issue. Might be a max_path problem?

Windows 10, Cmder, latest stable nvm-windows (https://github.com/coreybutler/nvm-windows/releases/tag/1.1.5).

1. install nvm-windows
2. > `nvm list`
`  * 8.9.4 (Currently using 64-bit executable)`
3. install 8.10
```
> nvm install 8.10
8.10.0
Downloading node.js version 8.10.0 (64-bit)...
Complete
Downloading npm version 5.6.0... Complete
Installing npm v5.6.0...2018/03/11 11:13:34 mkdir C:\Users\PetevanBlerk\AppData\Roaming\nvm\temp\nvm-npm\npm-5.6.0\node_modules\libnpx\node_modules\yargs\node_modules\read-pkg-up\node_modules\read-pkg\node_modules\load-json-file\node_modules\parse-json\node_modules\error-ex\node_modules\is-arrayish: The filename or extension is too long.
```
4. switch to 8.10
```
>  nvm use 8.10
8.10.0
Now using node v8.10.0 (64-bit)
```
5. Check version of node:
```
> node -v
v8.10.0
```
6. Confirm that **npm is missing** (consistent with install error above)
```
npm -v
'npm' is not recognized as an internal or external command,
operable program or batch file.
```
7. No npm in Appdata
![image](https://user-images.githubusercontent.com/5563334/37252726-2caa5272-251e-11e8-8233-258931be0fa1.png)

**Comment by petevb** [comment]
I've reinstalled nvm to d:\ instead of my roaming profile and can now install 8.10.0:
![image](https://user-images.githubusercontent.com/5563334/37272828-cc3174e0-25cf-11e8-961e-190ba27a5d57.png)

So suspect it is a MAX_PATH issue with nvm-windows.

**Comment by BouncingBit** [comment]
@petevb I can confirm, that this is a MAX_PATH issue as well.

**Comment by slimeygecko** [comment]
Have you tried the prerelease version? Using that version things installed
properly for me.
